src/test/spec/unified_runner/test_file.rs

```diff
@@ -0,0 +1,254 @@
+            let selection_criteria = SelectionCriteria::ReadPreference(read_preference.clone());
+            collection_options.selection_criteria = Some(selection_criteria);
+        }
+        collection_options.write_concern = self.write_concern.clone();
+        collection_options
+    }
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CollectionData {
+    pub collection_name: String,
+    pub database_name: String,
+    pub documents: Vec<Document>,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TestCase {
+    pub description: String,
+    pub run_on_requirements: Option<Vec<RunOnRequirement>>,
+    pub skip_reason: Option<String>,
+    pub operations: Vec<Operation>,
+    pub expect_events: Option<Vec<ExpectedEvents>>,
+    pub outcome: Option<Vec<CollectionData>>,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Operation {
+    pub name: String,
+    pub object: String,
+    pub arguments: Option<Document>,
+    pub expect_error: Option<ExpectError>,
+    pub expect_result: Option<Bson>,
+    pub save_result_as_entity: Option<String>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct ExpectedEvents {
+    pub client: String,
+    pub events: Vec<TestEvent>,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ExpectError {
+    pub is_error: Option<bool>,
+    pub is_client_error: Option<bool>,
+    pub error_contains: Option<String>,
+    pub error_code: Option<i32>,
+    pub error_labels_contain: Option<Vec<String>>,
+    pub error_labels_omit: Option<Vec<String>>,
+    pub expected_result: Option<Bson>,
+}
+
+impl ExpectError {
+    pub fn verify_result(self, error: Error) {
+        if let Some(is_client_error) = self.is_client_error {
+            assert_eq!(is_client_error, !error.is_server_error());
+        }
+        if let Some(error_contains) = self.error_contains {
+            match &error.kind.code_and_message() {
+                Some((_, msg)) => assert!(msg.contains(&error_contains)),
+                None => panic!("error should include message field"),
+            }
+        }
+        if let Some(error_code) = self.error_code {
+            match &error.kind.code_and_message() {
+                Some((code, _)) => assert_eq!(*code, error_code),
+                None => panic!("error should include code"),
+            }
+        }
+        if let Some(error_labels_contain) = self.error_labels_contain {
+            for label in error_labels_contain {
+                assert!(error.labels().contains(&label));
+            }
+        }
+        if let Some(error_labels_omit) = self.error_labels_omit {
+            for label in error_labels_omit {
+                assert!(!error.labels().contains(&label));
+            }
+        }
+        if self.expected_result.is_some() {
+            panic!("Bulk write not implemented");
```

Comment:
We do have bulk write by means of `insert_many`. We do not, however, report the partial results of the insert many when it fails (RUST-260), so we don't need to verify `expected_result` matches. Perhaps replace this `if`/`panic` combo with a `TODO: RUST-260` comment?